### DISCUSSION TITLE
nnn

### DISCUSSION THREAD

**Discussion by DartMitai** [fish-shell/fish-shell#10231] [Q&A] [answered]
Hello everyone, there is a file nnn.fish with settings, how can I add it to the fish? I tried to put it in ~/.config/fish/conf.d but the NNN settings are not loaded...

**Comment by faho** [comment]
If this is about the [nnn file manager](https://github.com/jarun/nnn), you'd probably be better off asking the developers of that. It depends on what "nnn.fish" *is*.

The only thing I can find in their repo is [a completion script](https://github.com/jarun/nnn/blob/f2a8648861ff88e1c2d5a95ae4d76b7367e12c9b/misc/auto-completion/fish/nnn.fish#L4), which would only change what happens when you do `nnn <TAB>`, and you would typically place it in ~/.config/fish/completions.

**Reply by DartMitai** [reply]
```
export NNN_PLUG='i:imgview;t:preview-tabbed'
export NNN_FCOLORS='00'
set --export NNN_FIFO "/tmp/nnn.fifo"
```
here is a simple configuration file, changing the color of folders and connecting two plugins

**Reply by faho** [reply]
Yes, you would want to put that into ~/.config/fish/config.fish, or in ~/.config/fish/conf.d in a file named with a ".fish" extension, like "nnn.fish".

Then you'll want to check that `nnn` is what you think it is - running `type nnn` should tell you it's a command in a file somewhere, like /usr/bin/nnn.

Then you'll want to check that your config variables aren't overwritten somewhere, try `set --show NNN_PLUG NNN_FCOLORS NNN_FIFO`.

**Reply by DartMitai** [reply]
![image](https://github.com/fish-shell/fish-shell/assets/104931116/c07669d3-29e0-4d96-b591-2ee13e86d78a)
variables don't exist?
![image](https://github.com/fish-shell/fish-shell/assets/104931116/369a7aa8-bbc7-45f9-9896-dc133545a4ae)

**Reply by faho** [reply, marked as answer]
Okay. Try defining these variables with `set -gx` (global and exported) in config.fish.

Put this in there:

```fish
set -gx NNN_PLUG 'i:imgview;t:preview-tabbed'
set -gx NNN_FCOLORS '00'
set -gx NNN_FIFO "/tmp/nnn.fifo"
```

Then restart fish and try nnn.

If that works, try putting it into ~/.config/fish/conf.d/my_nnn.fish. If that works, rename it to nnn.fish. If that still works you had an issue with scoping.

**Reply by DartMitai** [reply]
it was necessary to restart the shell)) Thank you for your support, you are super!
